docs/error_codes.md

```diff
@@ -0,0 +1,19 @@
+# XLA Error codes
+
+This page is a list of all error codes emitted by the XLA compiler.
+
+- [E1000](./errors/error_1000.md)
+- [E1001](./errors/error_1001.md)
+- [E1002](./errors/error_1002.md)
+- [E03xx](./errors/error_03xx.md)
+- [E0100](./errors/error_0100.md)
+- [E0101](./errors/error_0101.md)
+- [E0102](./errors/error_0102.md)
+- [E2xxx](./errors/error_2xxx.md)
+- [E11xx](./errors/error_11xx.md)
```

Comment:
Similar to above about adding empty pages. If we're not going to introduce empty pages, we should not be adding any links yet